common/io_exerciser: Ensure empty stringstream in ceph_test_rados_io_sequence RadosIO between ReST calls

JonBailey1993 opened the pull request:
common/io_exerciser: Ensure empty stringstream in ceph_test_rados_io_sequence RadosIO between ReST calls

Fixes minor issue introduced in #60330 where the stringstream was not always empty prior to sending an osd command when using the --testrecovery flag. This fixes the issue and prevents it sending malformed/incorrect json in these cases.

Signed-off-by: JonBailey1993 [jonathan.bailey1@ibm.com](mailto:jonathan.bailey1@ibm.com)

<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced (3383af5dd1a1057baa1dfaae2840b68a35ac9265)
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>

<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

JonBailey1993 commented:
@ronen-fr - I have just added you as a reviewer as this is only a small change as a follow up fix to my previous change (#60330) which you kindly reviewed

ronen-fr commented:
> @ronen-fr - I have just added you as a reviewer as this is only a small change as a follow up fix to my previous change (#60330) which you kindly reviewed

Just note that you have added me as an assignee, not a reviewer... :-)

ronen-fr commented on src/common/io_exerciser/RadosIo.cc:
```diff
@@ -11,6 +11,30 @@
+int send_mon_command(S& s, librados::Rados& rados, const char* name,
+                     ceph::buffer::list& inbl, ceph::buffer::list* outbl,
+                     Formatter* f) {
+  std::ostringstream oss;
```
order of lines?

JonBailey1993 replied:
I'm not sure I understand, can you expand on the question?

ronen-fr replied:
you may want to switch lines 30 & 31, as 'oss' is not required at 31

JonBailey1993 replied:
Ah, understood and agreed, will fix

JonBailey1993 commented:
> > @ronen-fr - I have just added you as a reviewer as this is only a small change as a follow up fix to my previous change (#60330) which you kindly reviewed
> 
> Just note that you have added me as an assignee, not a reviewer... :-)

Oops, fixed haha :-)

ronen-fr commented:
jenkins test make check arm64

JonBailey1993 commented:
jenkins test make check
